ledger-core/virtual/integration/method_test.go

```diff
@@ -348,867 +390,540 @@ func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
+func TestVirtual_CallOtherMethodInObject(t *testing.T) {
+	t.Log("C5116")
+	table := []struct {
+		name        string
+		stateSender contract.MethodIsolation
+	}{
+		{
+			name:        "ordered A.Foo calls unordered A.Bar",
+			stateSender: tolerableFlags(),
+		}, {
+			name:        "unordered A.Foo calls unordered A.Bar",
+			stateSender: intolerableFlags(),
+		},
+	}
+	for _, test := range table {
+		t.Run(test.name, func(t *testing.T) {
 
-		classA        = gen.UniqueGlobalRef()
-		outgoingA     = server.RandomLocalWithPulse()
-		objectAGlobal = reference.NewSelf(outgoingA)
+			mc := minimock.NewController(t)
 
-		classB        = gen.UniqueGlobalRef()
-		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
 
-		outgoingCallRef = gen.UniqueGlobalRef()
-	)
+			logger := inslogger.FromContext(ctx)
 
-	Method_PrepareObject(ctx, server, payload.Ready, objectBGlobal)
+			runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+				return execution.Request.CallSiteMethod
+			})
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+			server.IncrementPulseAndWaitIdle(ctx)
 
-	// add ExecutionMocks to runnerMock
-	{
-		outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).CallMethod(objectBGlobal, classB, "Foo", []byte("123"))
-		objectAResult := requestresult.New([]byte("finish A.New"), objectAGlobal)
-		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
-		objectAExecutionMock := runnerMock.AddExecutionMock("New")
-		objectAExecutionMock.AddStart(
-			func(ctx execution.Context) {
-				logger.Debug("ExecutionStart [A.New]")
-				require.Equal(t, classA, ctx.Request.Callee)
-				require.Equal(t, outgoingA, ctx.Request.CallOutgoing)
-			},
-			&execution.Update{
-				Type:     execution.OutgoingCall,
-				Error:    nil,
-				Outgoing: outgoingCall,
-			},
-		)
-		objectAExecutionMock.AddContinue(
-			func(result []byte) {
-				logger.Debug("ExecutionContinue [A.New]")
-				require.Equal(t, []byte("finish B.Foo"), result)
-			},
-			&execution.Update{
-				Type:   execution.Done,
-				Result: objectAResult,
-			},
-		)
+			var (
+				class         = gen.UniqueGlobalRef()
+				objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+			)
 
-		runnerMock.AddExecutionMock("Foo").AddStart(
-			func(ctx execution.Context) {
-				logger.Debug("ExecutionStart [B.Foo]")
-				require.Equal(t, objectBGlobal, ctx.Request.Callee)
-				require.Equal(t, objectAGlobal, ctx.Request.Caller)
-				require.Equal(t, []byte("123"), ctx.Request.Arguments)
-			},
-			&execution.Update{
-				Type:   execution.Done,
-				Result: requestresult.New([]byte("finish B.Foo"), objectBGlobal),
-			},
-		)
+			Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+
+			outgoingCall := execution.NewRPCBuilder(gen.UniqueGlobalRef(), objectAGlobal).CallMethod(objectAGlobal, class, "Bar", []byte{})
```

Comment:
add arguments and checks for them, please

Reply:
done